cmd/kubeadm/app/constants/constants.go

```diff
@@ -334,6 +334,9 @@ const (
 	// CoreDNSVersion is the version of CoreDNS to be deployed if it is used
 	CoreDNSVersion = "1.2.6"
 
+	// MinimumKubernetesVersion specifies minimum Kubernetes version supported by kubeadm
+	MinimumKubernetesVersion = "v1.12.0"
```

Comment:
We can probably trim this to `v1.12`, so we can cover more cases with this PR (like `v1.12.5`).

Reply:
I'd prefer to do it in a follow-up PR as this one is already big and could cause rebase conflicts if not merged soon. wdyh?

Reply:
better to have it in the same PR, IMO.

Reply:
ok, will do

Reply:
but i wonder what other cases we need to do?

Reply:
@rosti, @neolit123 Would something like this be acceptable:
```
// MinK8sVerMajorMinor specifies <major>.<minor> part of the minimum Kubernetes version supported by Kubeadm

MinK8sMajorMinor = "v1.12"

// MinimumKubernetesVersion specifies minimum Kubernetes version supported by kubeadm
MinimumKubernetesVersion = MinK8sVerMajorMinor + ".0"
```

Then we can use MinK8sVerMajorMinor constant in the rest of v1.12* strings.

Reply:
on each version bump we need to send a PR that does something like this:
https://github.com/kubernetes/kubernetes/pull/68649/files

the idea is to reduce this to a change in a single file - constants.go.

Reply:
I think I understood the idea. I don't understand why it has to be implemented as one PR.

Reply:
my vote would be for:
```
MinimumKubernetesVersionWithoutPatch = ...
MinimumKubernetesVersion = ...
```

in terms of the example change in:
https://github.com/kubernetes/kubernetes/pull/68649/files

we might as well add more constants for the sake of making it more easy.
e.g. Minor + 1

that's my 2c.

Reply:
I was thinking about having all changes, that involve `1.12.x` in a single PR (currently we replaced only `1.12.0` and left some `1.12.x` stuff, where x != 0, for another PR).

I am OK with moving `1.13` and `1.14` stuff in other PR(s). That way it will be more reviewable.

Reply:
@rosti 

> currently we replaced only 1.12.0 and left some 1.12.x stuff

Nope, almost everything is covered with MinimumKubernetesVersion and MinimumKubernetesVersionWithoutPatch. You probably missed my updates.

Below is what's left currently. Can you point me out what of this should be included into this PR?
```
$ grep -r v1.12 . | grep -v ':\s*\/\/'
Binary file ./kubeadm matches
./app/constants/constants.go:	MinimumKubernetesVersionWithoutPatch = "v1.12"
Binary file ./app/cmd/phases/phases.test matches
./app/cmd/phases/util_test.go:			input:  "v1.12.0",
./app/cmd/phases/util_test.go:			output: "v1.12.0",
./app/cmd/config_test.go:				":v1.12.1",
./app/cmd/config_test.go:				kubernetesVersion: v1.12.1
./app/componentconfigs/config_test.go:	k8sVersion := version.MustParseGeneric("v1.12.0")
./app/phases/uploadconfig/uploadconfig_test.go:					KubernetesVersion: "v1.12.10",
Binary file ./app/util/upgrade/upgrade.test matches
./app/util/config/testdata/conversion/master/v1alpha3.yaml:kubernetesVersion: v1.12.2
./app/util/config/testdata/conversion/master/v1alpha3.yaml:unifiedControlPlaneImage: "k8s.gcr.io/hyperkube:v1.12.2"
./app/util/config/testdata/conversion/master/v1beta1.yaml:kubernetesVersion: v1.12.2
./app/util/config/testdata/conversion/master/internal.yaml:KubernetesVersion: v1.12.2
./app/util/config/common.go:		"kubeadm.k8s.io/v1alpha2": "v1.12",
./app/util/version_test.go:			remoteVersion: "v1.12.0",
./app/util/version_test.go:			remoteVersion: "v1.12.1",
./app/util/version_test.go:			output:        "v1.12.1",
./app/util/version_test.go:			clientVersion: "v1.12.0",
./app/util/version_test.go:			remoteVersion: "v1.12.0",
./app/util/etcd/etcd.go:				klog.V(1).Infoln("etcd manifest created by kubeadm v1.12")
./app/apis/kubeadm/v1alpha3/conversion_test.go:				KubernetesVersion:        "v1.12.2",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				UnifiedControlPlaneImage: "k8s.gcr.io/hyperkube:v1.12.2",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				KubernetesVersion:        "v1.12.0",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				UnifiedControlPlaneImage: "k8s.gcr.io/hyperkube:v1.12.2",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				KubernetesVersion:        "v1.12.2",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				UnifiedControlPlaneImage: "k8s.gcr.io/hyperkube:v1.12.2",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				KubernetesVersion:        "v1.12.2",
./app/apis/kubeadm/v1alpha3/conversion_test.go:				UnifiedControlPlaneImage: "k8s.gcr.io/otherimage:v1.12.2",
```

Reply:
Sorry @bart0sh , I missed the updates. Will take a look.